File: Programmers/NaverHackday2020/Programming2.java
package com.java.Programmers.NaverHackday2020;

import java.util.LinkedHashMap;
import java.util.LinkedList;
import java.util.List;
import java.util.Map;

public class Programming2 {
    public static void main(String[] args) {
        System.out.println(solution(new String[]{"A B C D", "A D", "A B D", "B D"}, 2));
        System.out.println(solution(new String[]{"JAY", "JAY ELLE JAY MAY", "MAY ELLE MAY", "ELLE MAY", "ELLE ELLE ELLE", "MAY"}, 3));
    }

    public static int solution(String[] id_list, int k) {
        int answer = 0;
        Map<String, Integer> map = new LinkedHashMap<>();
        List<String> customerPerDay;
        for (int i = 0; i < id_list.length; i++) {
            String[] customers = id_list[i].split(" ");
            customerPerDay = new LinkedList<>();
            for (int j = 0; j < customers.length; j++) {
                String cur = customers[j];
                if (!customerPerDay.contains(cur))
                    customerPerDay.add(cur);
                else
                    continue;
                if (!map.containsKey(cur)) {
                    map.put(cur, 1);
                    answer++;
                } else {
                    if (map.get(cur) < k) {
                        map.replace(cur, map.get(cur) + 1);
                        answer++;
                    }
                }
            }
        }
        return answer;
    }
}

File: Programmers/Test210702/programming2.js
function solution(n) {
    let dp = new Array(10);
    for (let i = 0; i < 10; i++) {
        dp[i] = new Array(n + 1);
    }

    for (let i = 0; i < 10; i++) {
        for (let j = 0; j < n + 1; j++) {
            dp[i][j] = 0;
        }
    }

    for (let i = 0; i < 10; i++)
        dp[i][1] = 1;

    for (let d = 0; d < 10; d++) {
        for (let l = 2; l < n + 1; l++) {
            for (let x = 0; x < d; x++) {
                dp[d][l] += dp[x][l - 1];
            }
        }
    }
    console.log(dp)

    return dp.reduce((acc, cur) => acc + cur[n], 0);
}

console.log(solution(2))
console.log(solution(3))
console.log(solution(4))
console.log(solution(5))
console.log(solution(6))
console.log(solution(7))
console.log(solution(8))
console.log(solution(9))
console.log(solution(10))